File: app/src/main/java/com/qianlei/jiaowu/entity/Examination.java
package com.qianlei.jiaowu.entity;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import com.alibaba.fastjson.annotation.JSONField;

import lombok.Data;

/**
 * 考试信息的实体类
 *
 * @author qianlei
 */
@Data
@Entity
public class Examination {
    @PrimaryKey(autoGenerate = true)
    private int id;
    /**
     * 考试名称
     */
    @JSONField(name = "kcmc")
    @ColumnInfo(name = "name")
    private String name;
    /**
     * 考试地点
     */
    @JSONField(name = "cdmc")
    @ColumnInfo(name = "place")
    private String place;
    /**
     * 考试时间
     */
    @JSONField(name = "kssj")
    @ColumnInfo(name = "time")
    private String time;

    /**
     * 座位号
     */
    @JSONField(name = "zwh")
    @ColumnInfo(name = "seat_id")
    private String seatId;

    /**
     * 学年
     */
    @ColumnInfo(name = "year")
    private String year;

    /**
     * 学期
     */
    @ColumnInfo(name = "term")
    private String term;
}

File: app/src/main/java/com/qianlei/jiaowu/entity/Examination.kt
package com.qianlei.jiaowu.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.google.gson.annotations.SerializedName

/**
 * 考试信息的实体类
 *
 * @author qianlei
 */
@Entity
data class Examination(

        @PrimaryKey(autoGenerate = true)
        var id: Int = 0,
        /**
         * 考试名称
         */
        @SerializedName(value = "kcmc")
        @ColumnInfo(name = "name")
        var name: String? = null,
        /**
         * 考试地点
         */
        @SerializedName(value = "cdmc")
        @ColumnInfo(name = "place")
        var place: String? = null,
        /**
         * 考试时间
         */
        @SerializedName(value = "kssj")
        @ColumnInfo(name = "time")
        var time: String? = null,
        /**
         * 学年
         */
        @ColumnInfo(name = "year")
        var year: String? = null,
        /**
         * 学期
         */
        @ColumnInfo(name = "term")
        var term: String? = null
)